### DISCUSSION TITLE
Version progression on maintenance and release brances

### DISCUSSION THREAD

**Discussion by codekult** [semantic-release/semantic-release#3507] [Release workflow] [answered]
I need to provide some clarity (working on docs) around the release process for the rest of my team.

Let's suppose the following situation, we branch out a maintenance branch from 1.0.0 version, do some fixes and release 1.0.1 is created. Then we need to merge back these fixes to main branch, what would happen? Will a new version be generated (let's say 1.0.3 in main)? Is there any check that will avoid proceeding with the aforementioned process?

**Comment by travi** [comment, marked as answer]
you might be misunderstanding the purpose of maintenance release lines. the maintenance workflow is intended for situations where you have already released, for example, the next major version but still support the previous major version line. so, after releasing v2.0.0 from the main, stable release line, you can release a bug fix or security patch for v1.x.x. maintenance branches should not be merged back into main because the two release lines have diverged because of the fact that it is a maintenance branch. cherry picking such changes might be necessary, but that would normally happen in the other direction where you would fix a problem in the stable channel and cherry pick that change into the maintenance release line.

semantic-release encourages following continuous integration into the default branch and making most of your releases from that branch. you should not be trying to use branches for every release
